code/_onclick/other_mobs.dm

```diff
@@ -91,6 +91,10 @@
 		var/obj/item/clothing/gloves/G = gloves
 		if(istype(G) && G.Touch(A,0)) // for magic gloves
 			return
+	if (glasses)

```

Comment:
just do if istype(glasses) if it's already typecasted in carbon, otherwise make ranged_attack() on glasses able to return TRUE to interrupt the click.